Sprint-2/implement/tally.test.js

```diff
@@ -19,16 +19,26 @@ const tally = require("./tally.js");
 // Given a function called tally
 // When passed an array of items
 // Then it should return an object containing the count for each unique item
-
+test("tally When passed an array of items returns an object containing the count for each unique item", () =>{
+    expect(tally(["a", "b", "c"])).toEqual({a : 1, b : 1, c : 1});
+}); 
 // Given an empty array
 // When passed to tally
 // Then it should return an empty object
-test.todo("tally on an empty array returns an empty object");
+test("tally on an empty array returns an empty object", () =>{
+    expect(tally([])).toEqual({});
+});
 
 // Given an array with duplicate items
 // When passed to tally
 // Then it should return counts for each unique item
+test("tally When passed an array with duplicate items returns counts for each unique item", () =>{
+    expect(tally(["a", "b", "c", "a", "a", "c"])).toEqual({a : 3, b : 1, c : 2});
+});
 
 // Given an invalid input like a string
 // When passed to tally
 // Then it should throw an error
+test("tally When passed an invalid input like a string Then it should throw an error", () =>{
+    expect(tally("hello")).toThrow("Input must be an array");
```

Comment:
This is really close, but not _quite_ right. We haven't actually taught about `throw`ing in the course, but it's useful to know about! But right now this test fails, and we don't like failing tests :)

When using jest, if you use `toThrow`, you need to pass a _function_ to `expect` rather than just a value - if you rewrite this as:
```suggestion
    expect(() => {tally("hello") }).toThrow("Input must be an array");
```

you should see a more useful failure message - can you make this change, and then fix the test so it passes?

Your actual code works fine, but the test code here isn't quite correct :)